docs/arch/adr-005-profiles-directories.md

```diff
@@ -0,0 +1,175 @@
+# Search order for `profiles.yml`
+
+## Context
+dbt has a solid foundation of [convention over configuration](https://en.wikipedia.org/wiki/Convention_over_configuration) (CoC), and this proposal would lean into this further.
+
+### Current behavior
+dbt needs a `profiles.yml` configuration file for database connection info. I believe the current search order is:
+1. `--profiles-dir` option
+1. `DBT_PROFILES_DIR` environment variable
+1. `~/.dbt/` directory
+
+dbt also needs a `dbt_project.yml`. The current order of precedence is:
+1. `--project-dir` option
+1. current working directory
+
+### Desired behavior
+Search order for `profiles.yml`:
+1. `--profiles-dir` option
+1. `DBT_PROFILES_DIR` environment variable
+1. current working directory (**NEW**)
+1. `~/.dbt/` directory
+
+Search order for `dbt_project.yml`:
+1. `--project-dir` option
+1. `DBT_PROJECT_DIR` environment variable (**NEW**)
+1. current working directory
+
+### General design requirements
+There's two necessary pieces for dbt to use a profile to connect to a target database:
+- property definitions for the desired target database
+- secrets to plug into the definition slots
+
+There are two main design requirements in terms of discoverability and accessibility:
```

Comment:
Discoverable/Accessible in what sense? Like inside your project?

Reply:
Yep, "discoverable/accessible" was intended to correlate with being "safe to store in GitHub".